addons/mail/models/mail_message.py

```diff
@@ -949,6 +949,13 @@ def _message_format(self, fnames):
         self.check_access_rule('read')
         vals_list = self._read_format(fnames)
         safari = request and request.httprequest.user_agent.browser == 'safari'
+
+        thread_ids_by_model_name = dict()
+        for message in self:
+            if message.model and message.res_id:
+                thread_ids_by_model_name.setdefault(message.model, self.env[message.model])
+                thread_ids_by_model_name[message.model] |= self.env[message.model].browse(message.res_id)
```

Comment:
The rename suggestion to `thread_ids` was only if you replaced this here by set of ids instead of set of records (since only the id is used afterwards now).